components/camel-dataset/src/main/java/org/apache/camel/component/dataset/DataSetConsumer.java

```diff
@@ -35,11 +35,13 @@ public class DataSetConsumer extends DefaultConsumer {
     private DataSetEndpoint endpoint;
     private Processor reporter;
     private ExecutorService executorService;
+    private boolean dataSetIndex;
```

Comment:
let's make it final

Reply:
Another remark, don't you believe that `dataSetIndexOff` or something similar could be more appropriate as name?

Reply:
+1. Those are good remarks.

Reply:
Or if we take directly the negation, it could rather be `withHeader`

Reply:
> Or if we take directly the negation, it could rather be `withHeader`

I went with a similar one: `withIndexHeader` so it's more specific. Thanks for pointing it out.